src/code/client.ts

```diff
@@ -430,40 +465,44 @@ class CloudFileManagerClient {
 
   // must be called as a result of user action (e.g. click) to avoid popup blockers
   parseUrlAuthorizeAndOpen() {
-    if ((this.appOptions.hashParams != null ? this.appOptions.hashParams.fileParams : undefined) != null) {
+    if (this.appOptions.hashParams?.fileParams != null) {
       const [providerName, providerParams] = this.appOptions.hashParams.fileParams.split(':')
       const provider = this.providers[providerName]
       if (provider) {
         return provider.authorize(() => {
-          return this.openProviderFile(providerName(providerParams))
+          return this.openProviderFile(providerName, providerParams)
```

Comment:
Another bug made apparent by the TypeScript conversion: `providerName` is a string and so can't be called like a function. This presumably dates back to the CoffeeScript era where the code probably looked something like: `@openProviderFile providerName providerParams`.